### PR TITLE
Implement compactGE idiomatically

### PR DESCRIPTION
**Pull request by adithyaov**

### CODE REVIEW

File: src/Streamly/Internal/Data/Array/Stream/Mut/Foreign.hs [hunk 1]
```diff
@@ -246,6 +246,55 @@ compactLEParserD n = ParserD.Parser step initial extract
     functionPath =
         "Streamly.Internal.Data.Array.Stream.Mut.Foreign.compactLEParserD"
 
+-- | Coalesce adjacent arrays in incoming stream to form bigger arrays of a
+-- minimum specified size. Note that if a all the arrays in the stream together
```
Comment (harendra-kumar) [inline src/Streamly/Internal/Data/Array/Stream/Mut/Foreign.hs]: typo "a all".

File: src/Streamly/Internal/Data/Array/Stream/Mut/Foreign.hs [hunk 2]
```diff
@@ -246,6 +246,55 @@ compactLEParserD n = ParserD.Parser step initial extract
...
+    => Int -> FL.Fold m (Array a) (Array a)
+compactGEFold n = Fold step initial extract
+
+    where
+
+    nBytes = n * sizeOf (undefined :: a)
+
+    initial =
+        return
+            $ if n <= 0
```
Comment (harendra-kumar) [inline src/Streamly/Internal/Data/Array/Stream/Mut/Foreign.hs:268]: `n == 0` should be a valid case in this.

File: src/Streamly/Internal/Data/Array/Stream/Mut/Foreign.hs [hunk 3]
```diff
@@ -246,6 +246,55 @@ compactLEParserD n = ParserD.Parser step initial extract
...
+    step Nothing arr =
+        return
+            $ let len = MArray.byteLength arr
+               in if len >= nBytes
+                  then FL.Done arr
+                  else FL.Partial (Just arr)
+    step (Just buf) arr = do
+        let len = MArray.byteLength buf + MArray.byteLength arr
+        buf1 <-
+            if MArray.byteCapacity buf < len
```
Comment (harendra-kumar) [inline src/Streamly/Internal/Data/Array/Stream/Mut/Foreign.hs:284]: Since we have collect `>= n` we can allocate at least `n` right away.

Comment (harendra-kumar) [reply src/Streamly/Internal/Data/Array/Stream/Mut/Foreign.hs:284]: In compactLE as well we can allocate equal to `n` to avoid repeated allocations.

**Review by harendra-kumar — CHANGES_REQUESTED** [review]

File: src/Streamly/Internal/Data/Array/Stream/Mut/Foreign.hs (hunk 3 above)
Comment (adithyaov) [reply src/Streamly/Internal/Data/Array/Stream/Mut/Foreign.hs:284]: Yeah.

Comment (adithyaov) [reply src/Streamly/Internal/Data/Array/Stream/Mut/Foreign.hs:284]: In compactLE, we do it already.

**Review by harendra-kumar — APPROVED** [review]

**Pull request merged**